Update plugin list sidebar behavior on tablets

kellyi opened the pull request:
## Overview

On viewports narrower than 991px (which included iPads in portrait mode), the prototype had shown only plugin icons along with a not-straightforward-to-dismiss tooltip pointing to the icon when one of the buttons had been tapped. 

This PR removes the tooltips altogether and updates the plugin list sidebar behavior for viewports < 991px:

- when a plugin is open and displaying content in its sidebar container, we now show only the icons in the plugin list sidebar
- when no plugins are displaying content in a sidebar container, we now show the icons and plugin titles in the sidebar

This works by toggling a CSS class on the `nav-app` class, which houses the plugin list. When plugin content is visible, `nav-app` now gets a `nav-app-narrow` class -- which is removed when no plugin content's visible. That class has rules within a `@media (max-width: 991px) {` CSS rule which hides the text for the plugin buttons.

## Screenshots

![img_0100](https://cloud.githubusercontent.com/assets/4165523/21509708/84d98446-cc59-11e6-9fd5-802e8f8e6e41.PNG)

![img_0106](https://cloud.githubusercontent.com/assets/4165523/21509713/9a6311b0-cc59-11e6-8c7c-5540c6e20940.PNG)

![img_0105](https://cloud.githubusercontent.com/assets/4165523/21509715/a67a0a44-cc59-11e6-82e3-13d879ff4dc0.PNG)

## Notes

To see selectable layers in regional-planning, make sure you've got the most recent commits from the plugin's `prototype` branch.

## Testing
- rebuild this branch in VS, then launch in the browser
- use the app as usual in your browser and verify that everything still works as before on a viewport > 990 px;
- use ngrok, iisexpress-proxy, or similar to make your version available to an iPad, then connect and check the site in both landscape and portrait mode
- in landscape mode, verify that everything still works as it did before and that the plugin buttons display the titles
- in portrait mode, verify that the plugin list sidebar now shows only icons when a plugin's content is visible, shows icons and titles when no plugin content is visible, and does not show any tooltips on taps
- try to get the plugin list sidebar out of sync with whether plugin content is visible. Do this by trying combinations of tapping a plugin to open it, selecting another one, selecting another one, closing/minimizing then reopening plugins in various sequences. Verify that the show/hide plugin titles behavior continues to work correctly.

Connects #771

kellyi commented:
Probably also worth ensuring that the plugin list sidebar updates correctly when the size of the viewport changes, as in when a tablet's turned from portrait to landscape mode. The expected behavior is that it should be sized correctly automatically on turning to landscape, as the `media(max-width)` rule will no longer apply.

mmcfarland commented on src/GeositeFramework/js/Plugin.js:
```diff
@@ -196,6 +196,14 @@ require(['use!Geosite',
+                    // Remove the `.nav-apps-narow` class if the
+                    // de-selecting the plugin has left no plugins
+                    // active.
+                    if ($('.sidebar').filter(function() {
```
I don't think we should be using the CSS properties for state in the app.  Instead, you can check if all plugins are  not `active`:
```js
if (this.collection.all({active:false})) {
    $('.nav-apps').removeClass('nav-apps-narrow');
}
````

kellyi replied:
Much better -- I'll make that change here and in line 422 referred to in the next comment.

mmcfarland commented on src/GeositeFramework/js/Plugin.js:
```diff
@@ -409,17 +417,28 @@ require(['use!Geosite',
-                }));
+                })),
+                sidebarNavSizedForTablet = !!$('.nav-apps-narrow'),
+                pluginContentHidden = $('.sidebar').filter(function() {
```
I _believe_ this is referring to all plugins, not just "this" plugin.  If so you, you can use the same model state from the previous comment, but you'll access the collection with `view.model.collection`.

mmcfarland commented on src/GeositeFramework/js/Plugin.js:
```diff
@@ -443,6 +462,17 @@ require(['use!Geosite',
             return view;
         }
 
+        function toggleTabletSidebarPluginText(visible) {
```
`toggle` implies that you can invert the current setting.  If this really requires a bool arg, I'd rename it to `setSideBarPluginTextVisibility`, and drop the `Tablet` since that really hasn't got anything to do with the function implementation.

kellyi commented:
Pushed a fixup which addresses the requested changes. I updated the function name to make it more clear what it was going to do, and adjusted the checks to read model/collection state rather than reading from the UI. 

However, it seemed like `active` wasn't quite the thing I needed to check on, as it seems like plugins which have been minimized yet are still on remain `active`. I think what it should be checking is whether any plugins are `visible`, and so I added that as part of the model state, then set and updated it accordingly. This may be already available in another way and, if so, it'd supplant needing to add a new key to keep track of in the state.

kellyi commented:
Ah, that makes sense. I checked out picky and it looks like the repo's README says it's deprecated and that people should consider using Backbone Select instead. I'll leave the `visible` property as part of the state here.

Thanks for your feedback on this! Going to squash & merge.

kellyi commented:
Squashed and rebased on `prototype`.
